mm2src/coins/z_coin/storage/blockdb/blockdb_idb_storage.rs

```diff
@@ -177,10 +171,11 @@ impl BlockDbImpl {
         let block_db = db_transaction.table::<BlockDbTable>().await?;
 
         // Fetch CompactBlocks block_db are needed for scanning.
+        let min = u32::from(from_height + 1);
         let mut maybe_blocks = block_db
             .cursor_builder()
             .only("ticker", &self.ticker)?
-            .bound("height", u32::from(from_height + 1), limit.unwrap_or(u32::MAX))
+            .bound("height", min, u32::MAX)
```

Comment:
`query_blocks_by_limit` seems not used anymore and likely has a bug checking `if block.height > limit`. Please either remove it or fix and use in `process_blocks_with_mode`.

Reply:
fix and use `query_blocks_by_limit`. thanks